reference.rst

```diff
@@ -13,3 +13,4 @@ Reference
    reference/conanfile
    reference/tools
    reference/extensions
+   reference/hooks
```

Comment:
I think it is better to put the hooks inside ``extensions``, not at the first level, together with the many other extensions: custom commands, custom plugins (compatibility, profile), etc. The hooks are probably not more important that custom commands in 2.0, and that would mean we need to put everything at the root level, and that would be too much.

Reply:
Done, kept only extensions/hooks